### PR TITLE
Task/hw 53264 add ui test transaction details

### PR DESCRIPTION
**Pull request by malaw-hw**

### CODE REVIEW

File: HyperwalletUISDK.xcodeproj/xcshareddata/xcschemes/Demo.xcscheme [hunk 1]
```diff
@@ -61,15 +61,16 @@
       debugDocumentVersioning = "YES"
       debugServiceExtension = "internal"
       allowLocationSimulation = "YES">
-      <MacroExpansion>
+      <BuildableProductRunnable
```
Comment (oleksii-lubianyi-epam) [inline HyperwalletUISDK.xcodeproj/xcshareddata/xcschemes/Demo.xcscheme]: Please restore default settings

File: UITests/Responses/Receipts/ReceiptsForOneMonth.json [hunk 2]
```diff
@@ -66,6 +59,25 @@
...
                 "payeeName": "Shaoyun Yang"
             }
+        },
+        {
+            "journalId": "3051579",
+            "type": "PAYMENT",
+            "createdOn": "2019-05-03T17:08:58",
+            "entry": "CREDIT",
+            "sourceToken": "act-12345",
+            "destinationToken": "usr-fa76a738-f43d-48b9-9a7a-7048d44a5d2d",
```
Comment (oleksii-lubianyi-epam) [inline UITests/Responses/Receipts/ReceiptsForOneMonth.json]: Should real token appear here?

File: UITests/Responses/Receipts/ReceiptsForOneMonth.json [hunk 3]
```diff
@@ -66,6 +59,25 @@
...
+            "amount": "20.00",
+            "fee": "2.25",
+            "currency": "USD",
+            "details": {
+                "clientPaymentId": "8OxXefx5",
+                "payeeName": "A Person",
+                "website": "https://api.sandbox.hyperwallet.com",
```
Comment (oleksii-lubianyi-epam) [inline UITests/Responses/Receipts/ReceiptsForOneMonth.json]: Should be "https://localhost", I think

File: UITests/Helper/Receipts/TransactionDetails.swift [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    // Back button
+    var backButton: XCUIElement
+
+    var app: XCUIApplication
+
+    init(app: XCUIApplication) {
+        self.app = app
+        navigationBar = app.navigationBars["Transaction Details"]
+        detailHeaderTitle = app.navigationBars["Transaction Details"].otherElements["Transaction Details"]
+        receiptdetailtableviewTable = app.tables["receiptDetailTableView"]
```
Comment (syyang-hw) [inline UITests/Helper/Receipts/TransactionDetails.swift]: use camelcase receiptDetailTableView

File: UITests/Receipts/TransactionDetailTests.swift [hunk 5]
```diff
@@ -0,0 +1,201 @@
+import XCTest
+
+class TransactionDetailTests: BaseTests {
+    var transactDetails: TransactionDetails!
```
Comment (syyang-hw) [inline UITests/Receipts/TransactionDetailTests.swift]: use private accessor if it's only being used in this class.

File: UITests/Receipts/TransactionDetailTests.swift [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+    let currency = "USD"
```
Comment (syyang-hw) [inline UITests/Receipts/TransactionDetailTests.swift]: Same as above

File: UITests/Receipts/TransactionDetailTests.swift [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+        let transactionDetailHeaderLabel = transactDetails.detailHeaderTitle
+        waitForNonExistence(transactionDetailHeaderLabel)
+        if #available(iOS 12, *) {
+            verifyPayment(payment: "Payment\nMay 24, 2019", amount: "+6.00\n\(currency)")
+        } else {
+            verifyPayment(payment: "Payment May 24, 2019", amount: "+6.00 \(currency)")
+        }
+
+        // DETAILS Section
+        // note: iOS 10 Shows Date's time as 16:16 PM
```
Comment (syyang-hw) [inline UITests/Receipts/TransactionDetailTests.swift]: Is this notes still valid?

File: UITests/Receipts/TransactionDetailTests.swift [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+
+        // skip the test on iOS 10 as simulator time is not correct - possible XCode's bug
+        if #available(iOS 11, *) {
+            XCTAssertTrue(dateLabel.exists)
+            XCTAssertTrue(date.exists)
+        }
+    }
+
+    // FEE section verification
+    private func verifyFeeSection(amount: String, fee: String, trans: String) {
```
Comment (syyang-hw) [inline UITests/Receipts/TransactionDetailTests.swift]: Can you change trans -> transaction?

File: UITests/Receipts/TransactionDetailTests.swift [hunk 9]
```diff
@@ -0,0 +1,201 @@
...
+    func testReceiptDetail_verifyTransactionReceiptNoNoteSectionAndFeeLabel() {
+        openupReceiptsListScreenForFewMonths(isOneMonth: true)
+        transactDetails.openReceipt(row: 2)
+        let transactionDetailHeaderLabel = transactDetails.detailHeaderTitle
+        waitForNonExistence(transactionDetailHeaderLabel)
+
+        // Assert No Note and Fee sections
+        let noteSection = transactDetails.noteSectionLabel
+        let feeLabel = transactDetails.feeLabel
+        XCTAssertTrue(!noteSection.exists)
```
Comment (syyang-hw) [inline UITests/Receipts/TransactionDetailTests.swift]: You can simply use XCTAssertFalse

File: UITests/Receipts/TransactionDetailTests.swift [hunk 10]
```diff
@@ -0,0 +1,201 @@
...
+        app.tables.cells.containing(.staticText, identifier: "List Receipts").element(boundBy: 0).tap()
+        waitForNonExistence(spinner)
+    }
+
+    private func verifyPayment(payment: String, amount: String) {
+        let receiptdetailtableviewTable = transactDetails.receiptdetailtableviewTable
+        let paymentlabel = receiptdetailtableviewTable.staticTexts["ListReceiptTableViewCellTextLabel"].label
+        let amountlabel = receiptdetailtableviewTable.staticTexts["ListReceiptTableViewCellDetailTextLabel"].label
+
+        XCTAssertTrue(paymentlabel == payment)
```
Comment (syyang-hw) [inline UITests/Receipts/TransactionDetailTests.swift]: Please use XCAssertEqual

File: UITests/Responses/Receipts/ReceiptsForOneMonth.json (hunk 2 above)
Comment (syyang-hw) [reply UITests/Responses/Receipts/ReceiptsForOneMonth.json]: Use dummy token

File: UITests/Responses/Receipts/ReceiptsForOneMonth.json (hunk 3 above)
Comment (syyang-hw) [reply UITests/Responses/Receipts/ReceiptsForOneMonth.json]: @oleksii-lubianyi-epam is right

**Pull request merged**